engine/src/util.cpp

```diff
@@ -3182,6 +3186,110 @@ bool MCU_random_bytes(size_t p_bytecount, MCDataRef& r_bytes)
 
 ///////////////////////////////////////////////////////////////////////////////
 
+// AL-2015-02-06: [[ SB Inclusions ]] Add utility functions for module loading where
+//  p_module can be a universal module name, where a mapping from module names to
+// relative paths has been provided.
+// SN-2015-02-23: [[ Broken Win Compilation ]] Use void*, as the function is imported
+//  as extern in revbrowser/src/cefshared.h - where MCSysModuleHandle does not exist
+void* MCU_loadmodule(const char *p_module)
+{
+    MCSysModuleHandle t_handle;
+    t_handle = nil;
+#if defined(_MACOSX) || defined(_MAC_SERVER)
+    t_handle = (MCSysModuleHandle)NSAddImage(p_module, NSADDIMAGE_OPTION_RETURN_ON_ERROR | NSADDIMAGE_OPTION_WITH_SEARCHING);
+    if (t_handle != nil)
+        return t_handle;
+    // MM-2014-02-06: [[ LipOpenSSL 1.0.1e ]] On Mac, if module cannot be found then look relative to current executable.
+    uint32_t t_buffer_size;
+    t_buffer_size = 0;
+    _NSGetExecutablePath(NULL, &t_buffer_size);
+    char *t_module_path;
+    t_module_path = (char *) malloc(t_buffer_size + strlen(p_module) + 1);
+    if (t_module_path != NULL)
+    {
+        if (_NSGetExecutablePath(t_module_path, &t_buffer_size) == 0)
+        {
+            char *t_last_slash;
+            t_last_slash = t_module_path + t_buffer_size;
+            for (uint32_t i = 0; i < t_buffer_size; i++)
+            {
+                if (*t_last_slash == '/')
+                {
+                    *(t_last_slash + 1) = '\0';
+                    break;
+                }
+                t_last_slash--;
+            }
+            strcat(t_module_path, p_module);
+            t_handle = (MCSysModuleHandle)NSAddImage(t_module_path, NSADDIMAGE_OPTION_RETURN_ON_ERROR | NSADDIMAGE_OPTION_WITH_SEARCHING);
+        }
+        free(t_module_path);
+        // AL-2015-02-17: [[ SB Inclusions ]] Return the handle if found here.
+        if (t_handle != nil)
+            return t_handle;
+    }
+#endif
+
+    MCAutoStringRef t_path;
+    
+    if (!MCdispatcher || !MCdispatcher -> fetchlibrarymapping(p_module, &t_path))
+    {
+        if (!MCStringCreateWithCString(p_module, &t_path))
+            return nil;
+    }
+
+    t_handle = MCS_loadmodule(*t_path);
+    
+    if (t_handle != nil)
+        return t_handle;
+    
+    MCAutoStringRef t_filename;
+    if (MCStringGetCharAtIndex(*t_path ,0) == '/')
+    {
+        if (!MCStringCopy(*t_path, &t_filename))
+            return nil;
+    }
+    else
+    {
+        uindex_t t_last_slash_index;
+        if (!MCStringLastIndexOfChar(*t_path, '/', UINDEX_MAX, kMCStringOptionCompareExact, t_last_slash_index))
+            t_last_slash_index = MCStringGetLength(*t_path);
+
+        if (!MCStringFormat(&t_filename, "%*@/%s", MCRangeMake(0, t_last_slash_index), MCcmd, *t_path))
+            return nil;
```

Comment:
There are a couple of issues here: MCStringLastIndexOfChar should be called on MCcmd (and MCStringGetLength).
The format string for MCStringFormat is wrong - the %s should be a %@. Also the MCRange parameter for MCStringFormat has to be a pointer to a range rather than a range.